### PR TITLE
Add new rules to PMD - exceptions

### PR DESCRIPTION
**Pull request by kanghj**
Added rules for exceptions other than SignatureDeclareThrowsException, AvoidCatchingGenericException, AvoidThrowingRawExceptionTypes. Don't know if those 3 rules are worth the effort

### CODE REVIEW

File: src/test/java/teammates/test/cases/common/ActivityLogEntryTest.java [hunk 1]
```diff
@@ -47,6 +47,11 @@ public void testActivityLogEntryClass() {
     public void testGetActionName(){
         assertEquals("instructorCourse", ActivityLogEntry.getActionName("/page/instructorCourse"));
         assertEquals("instructorCourse", ActivityLogEntry.getActionName("/page/instructorCourse?user=x"));
-        assertEquals("error in getActionName for requestUrl : instructorCourse", ActivityLogEntry.getActionName("instructorCourse"));
+        try {
+            ActivityLogEntry.getActionName("instructorCourse");
+            signalFailureToDetectException("getActionName should throw an exception if an action cannot be retrieved");
+        } catch (Exception e) {
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/common/ActivityLogEntryTest.java]: I believe this is specifically `ArrayIndexOutOfBoundsException`. Try not to catch a generic one.

File: src/main/java/teammates/logic/backdoor/BackDoorLogic.java [hunk 2]
```diff
@@ -392,15 +392,21 @@ private FeedbackSessionAttributes cleanSessionData(FeedbackSessionAttributes ses
...
+                    qnNumber);
+            if (question == null) {
+                throw new EntityDoesNotExistException("question has not persisted yet");
+            }
             response.feedbackQuestionId = 
-                feedbackQuestionsLogic.getFeedbackQuestion(
-                        response.feedbackSessionName, response.courseId,
-                        qnNumber).getId();
+                question.getId();
+
```
Comment (wkurniawan07) [inline src/main/java/teammates/logic/backdoor/BackDoorLogic.java]: Is this change required by the PMD rules?

Comment (kanghj) [inline src/main/java/teammates/logic/backdoor/BackDoorLogic.java]: Judging from history, this is where the NPE that <code>AdminInstructorAccountAddAction</code> was catching came from. It's not required by PMD, but the behaviour in <code>AdminInstructorAccountAddAction</code> would change if I didn't modify it to throw a EntityDoesNotExistException if its null

Comment (wkurniawan07) [inline src/main/java/teammates/logic/backdoor/BackDoorLogic.java]: Ok then. One small request: put the `question.getId()` in the same line as the previous one. It looks odd to break very short lines.

**Comment by wkurniawan07** [comment]
Two small requests and it's good to go.
